integration-libs/segment-refs/root/http-interceptors/occ-segment-refs.interceptor.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+} from '@angular/common/http';
+import { Injectable, inject, isDevMode } from '@angular/core';
+import { Observable } from 'rxjs';
+import { LoggerService, OccEndpointsService, WindowRef } from '@spartacus/core';
+import { SegmentRefsConfig } from '../config/segment-refs-config';
+
+@Injectable({ providedIn: 'root' })
+export class OccSegmentRefsInterceptor implements HttpInterceptor {
+  private segmentRefs?: string | null;
+  private requestHeader?: string;
+  protected readonly SEGMENT_REFS_KEY = 'segment-refs';
+
+  protected logger = inject(LoggerService);
+
+  constructor(
+    protected config: SegmentRefsConfig,
+    protected occEndpoints: OccEndpointsService,
+    protected winRef: WindowRef
+  ) {
+    const url = this.winRef.location.href ?? '';
+    const queryParams = new URLSearchParams(url.substring(url.indexOf('?')));
+    this.segmentRefs = queryParams.get('segmentrefs');
+    if (this.segmentRefs) {
+      this.winRef.localStorage?.setItem(
+        this.SEGMENT_REFS_KEY,
+        this.segmentRefs
+      );
+    } else {
+      this.segmentRefs = this.winRef.localStorage?.getItem(
+        this.SEGMENT_REFS_KEY
+      );
+    }
+    if (this.winRef.isBrowser()) {
+      if (!this.config.segmentRefs?.httpHeaderName && isDevMode()) {
+        this.logger.warn(`There is no httpHeaderName configured in Segment`);
+      }
+      this.requestHeader =
+        this.config.segmentRefs?.httpHeaderName?.toLowerCase();
+    }
+  }
```

Comment:
For easier extensibility for customers, we should never store any logic in constructor body - because it's not extendable/replacable.
Let's move the whole body to a protected method `initialize()` and then call it in the `constructor()`

Reply:
done